homeassistant/components/minecraft_server/__init__.py

```diff
@@ -0,0 +1,262 @@
+        self.description = None
+        self.version = None
+        self.protocol_version = None
+        self.latency_time = None
+        self.players_online = None
+        self.players_max = None
+        self.players_list = None
+
+        # Dispatcher signal name
+        self.signal_name = f"{SIGNAL_NAME_PREFIX}_{self.unique_id}"
+
+        # Callback for stopping periodic update.
+        self._stop_periodic_update = None
+
+    def start_periodic_update(self) -> None:
+        """Start periodic execution of update method."""
+        self._stop_periodic_update = async_track_time_interval(
+            self._hass, self.async_update, timedelta(seconds=SCAN_INTERVAL)
+        )
+
+    def stop_periodic_update(self) -> None:
+        """Stop periodic execution of update method."""
+        self._stop_periodic_update()
+
+    async def async_check_connection(self) -> None:
+        """Check server connection using a 'ping' request and store result."""
+        try:
+            await self._hass.async_add_executor_job(
+                self._mc_status.ping, self._RETRIES_PING
+            )
+            self.online = True
+        except IOError as error:
+            _LOGGER.debug("Error occurred while trying to ping the server: %s", error)
+            self.online = False
+
+    async def async_update(self, now: datetime = None) -> None:
+        """Get server data from 3rd party library and update properties."""
+        # Check connection status.
+        server_online_old = self.online
+        await self.async_check_connection()
+        server_online = self.online
+
+        # Inform user once about connection state changes if necessary.
+        if server_online_old and not server_online:
+            _LOGGER.warning("Connection to server lost")
+        elif not server_online_old and server_online:
+            _LOGGER.info("Connection to server (re-)established")
+
+        # Try to update the server data if server is online.
+        if server_online:
+            try:
+                await self._async_status_request()
+            except OSError as error:
+                _LOGGER.debug(
+                    "Error occurred while trying to update the server data: %s", error
+                )
+        else:
+            # Set all properties except description and version information to
+            # unknown until server connection is established again.
+            self.players_online = None
+            self.players_max = None
+            self.players_list = None
+            self.latency_time = None
+
+        # Notify sensors about new data.
+        async_dispatcher_send(self._hass, self.signal_name)
+
+    async def _async_status_request(self) -> None:
+        """Request server status and update properties."""
+        try:
+            status_response = await self._hass.async_add_executor_job(
+                self._mc_status.status, self._RETRIES_STATUS
+            )
+        except IOError as error:
```

Comment:
Why do we catch the exception just to raise it again?

Reply:
You're right. I will remove the try/catch block in `async_update()` and handle the OSError only in `_async_status_request()`.